starknet-rpc-test/src/lib.rs

```diff
@@ -214,3 +216,29 @@ impl MadaraClient {
         Ok(response.status().is_success())
     }
 }
+
+pub struct LegacyProgramWrapper {
+    pub legacy_program: LegacyProgram,
+}
+
+impl<'de> Deserialize<'de> for LegacyProgramWrapper {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = serde_json::Value::deserialize(deserializer)?;
+
+        // Ensure the required fields are present in the JSON data
+        let mut json_obj = value.as_object().ok_or_else(|| de::Error::custom("Expected JSON object"))?.clone();
+
+        // If 'main_scope' field is missing, set it to the default value
+        if !json_obj.contains_key("main_scope") {
+            json_obj.insert("main_scope".to_string(), serde_json::Value::String("__main__".to_string()));
+        }
+
+        // Deserialize the modified JSON data into LegacyProgram
+        let legacy_program = serde_json::from_value(serde_json::Value::Object(json_obj)).map_err(de::Error::custom)?;
+
+        Ok(LegacyProgramWrapper { legacy_program })
+    }
+}
```

Comment:
This comes from another PR.

Can you rebase please. Right now I can't tell what are your changes and what comes form the branch you merged into yours